### PR TITLE
feat: add json schema functionality to migrate contract

### PR DESCRIPTION
**Pull request by poomthiti**

### CODE REVIEW

File: src/lib/services/codeService.ts [hunk 1]
```diff
@@ -260,3 +272,18 @@ export const useCodeListCountByWalletAddress = (
     }
   );
 };
+
+export type LcdCodeInfoSuccessCallback = (data: CodeIdInfoResponse) => void;
+
+export const useLcdCodeInfo = (
```
Comment (songwongtp) [inline src/lib/services/codeService.ts:278]: `useLCDCodeInfo`

**Review by evilpeach — CHANGES_REQUESTED** [review]
1. Please adjust the width of the input field. It should span 100%
image
2. Move all files in `/common` outside and export all in `index.ts`

File: src/lib/services/codeService.ts [hunk 2]
```diff
@@ -260,3 +272,18 @@ export const useCodeListCountByWalletAddress = (
     }
   );
 };
+
+export type LcdCodeInfoSuccessCallback = (data: CodeIdInfoResponse) => void;
```
Comment (evilpeach) [inline src/lib/services/codeService.ts:276]: LCD

File: src/lib/services/codeService.ts [hunk 3]
```diff
@@ -260,3 +272,18 @@ export const useCodeListCountByWalletAddress = (
...
+
+export const useLcdCodeInfo = (
+  codeId: string,
+  options?: Parameters<typeof useQuery<CodeIdInfoResponse>>[2]
```
Comment (evilpeach) [inline src/lib/services/codeService.ts:280]: `options?: Omit<UseQueryOptions<CodeIdInfoResponse>, "queryKey">`

File: src/lib/pages/migrate/components/MigrateContract.tsx [hunk 4]
```diff
@@ -107,123 +127,172 @@ export const MigrateContract = ({
...
       contractAddress,
-      codeId: Number(codeId),
-      migrateMsg: JSON.parse(migrateMsg),
+      codeId,
+      currentInput,
       estimatedFee,
-      onTxSucceed: () => setProcessing(false),
-      onTxFailed: () => setProcessing(false),
-    });
+      broadcast,
```
Comment (evilpeach) [inline src/lib/pages/migrate/components/MigrateContract.tsx:153]: add `setProcessing`

File: src/lib/pages/migrate/components/MigrateContract.tsx [hunk 5]
```diff
@@ -107,123 +127,172 @@ export const MigrateContract = ({
...
+        </Heading>
+        <CodeSelectSection
+          name="codeId"
+          control={control}
+          status={status}
+          error={errors.codeId?.message}
+          onCodeSelect={(code: string) => {
+            setValue("codeId", code);
+          }}
+          setCodeHash={(data: Parameters<LcdCodeInfoSuccessCallback>[0]) => {
```
Comment (evilpeach) [inline src/lib/pages/migrate/components/MigrateContract.tsx:199]: Why don't use `CodeIdInfoResponse`

**Review by evilpeach — APPROVED** [review]

**Review by songwongtp — APPROVED** [review]

**Pull request merged**